crypto.go

```diff
@@ -699,8 +758,21 @@ func (s hkdfScheme) Expand(prk, info []byte, outLen int) []byte {
 	return out[:outLen]
 }
 
-func (s hkdfScheme) OutputSize() int {
-	return s.hash.Size()
+func (s hkdfScheme) LabeledExtract(salt []byte, label string, ikm []byte) []byte {
+	labeledIKM := append([]byte(rfcLabel+" "+label), ikm...)
+	return s.Extract(salt, labeledIKM)
+}
+
+func (s hkdfScheme) LabeledExpand(prk []byte, label string, info []byte, L uint16) []byte {
```

Comment:
It's not really useful to enforce the size of `L` using the type system.  As you can see, it causes a bunch of annoying casts to `uint16`, and since those casts are un-checked (they just throw away extra bits) you don't get any extra safety.  

Especially since the range of values is pretty fixed, I might be inclined to just take `int` there and `panic` if it's out of bounds.  In addition to avoiding casts, it actually provides the protection you're looking for.

Reply:
Great suggestion! I'll take it.